### PR TITLE
Support extra Ddl statements on create/update commands

### PR DESCRIPTION
**Pull request by benwulfe**
addresses https://github.com/GoogleCloudPlatform/google-cloud-dotnet/issues/1487

### CODE REVIEW

File: apis/Google.Cloud.Spanner.Data/Google.Cloud.Spanner.Data.IntegrationTests/Admin.cs [hunk 1]
```diff
@@ -53,5 +53,96 @@ public AdminTests(TestDatabaseFixture testFixture, ITestOutputHelper outputHelpe
                 await dropCommand.ExecuteNonQueryAsync().ConfigureAwait(false);
             }
         }
+
+        [Fact]
+        public async Task CreateWithExtrasDrop()
```
Comment (jskeet) [inline apis/Google.Cloud.Spanner.Data/Google.Cloud.Spanner.Data.IntegrationTests/Admin.cs]: Does "WithExtras" here refer to passing multiple commands into `CreateDdlCommand`?

File: apis/Google.Cloud.Spanner.Data/Google.Cloud.Spanner.Data.IntegrationTests/Admin.cs [hunk 2]
```diff
@@ -53,5 +53,96 @@ public AdminTests(TestDatabaseFixture testFixture, ITestOutputHelper outputHelpe
...
+        {
+            string dbName = "t_" + Guid.NewGuid().ToString("N").Substring(0, 28);
+
+            using (var connection = new SpannerConnection(_testFixture.NoDbConnectionString))
+            {
+                const string tableCreate = @"CREATE TABLE TX (
```
Comment (jskeet) [inline apis/Google.Cloud.Spanner.Data/Google.Cloud.Spanner.Data.IntegrationTests/Admin.cs]: So the "extra" statements are implicitly referring to the newly created database, yes? (Makes sense.)

File: apis/Google.Cloud.Spanner.Data/Google.Cloud.Spanner.Data/SpannerCommand.cs [hunk 3]
```diff
@@ -323,23 +323,42 @@ public override Task<int> ExecuteNonQueryAsync(CancellationToken cancellationTok
             if (spannerCommandTextBuilder.IsCreateDatabaseCommand)
             {
                 var parent = new InstanceName(spannerConnection.Project, spannerConnection.SpannerInstance);
-                var response = await databaseAdminClient.CreateDatabaseAsync(parent, commandText).ConfigureAwait(false);
+                var response = await databaseAdminClient.CreateDatabaseAsync(
```
Comment (jskeet) [inline apis/Google.Cloud.Spanner.Data/Google.Cloud.Spanner.Data/SpannerCommand.cs]: Can the multiple commands create multiple databases? What about "create table" (for a connection with a database) followed by "create database" (creating a separate one)? Hopefully we don't need to worry about these and the server will just Do The Right Thing.

File: apis/Google.Cloud.Spanner.Data/Google.Cloud.Spanner.Data/SpannerCommandTextBuilder.cs [hunk 4]
```diff
@@ -255,17 +255,24 @@ public static SpannerCommandTextBuilder CreateUpdateTextBuilder(string table)
         /// for executing a DDL statement.
         /// </summary>
         /// <param name="ddlStatement">The full DDL statement. Must not be null.</param>
+        /// <param name="extraDdlStatements"></param>
```
Comment (jskeet) [inline apis/Google.Cloud.Spanner.Data/Google.Cloud.Spanner.Data/SpannerCommandTextBuilder.cs]: Docs :)

File: apis/Google.Cloud.Spanner.Data/Google.Cloud.Spanner.Data/SpannerCommandTextBuilder.cs [hunk 5]
```diff
@@ -255,17 +255,24 @@ public static SpannerCommandTextBuilder CreateUpdateTextBuilder(string table)
...
         /// <returns>A <see cref="SpannerCommandTextBuilder"/> representing a <see cref="F:SpannerCommandType.Ddl"/> Spanner command.</returns>
-        public static SpannerCommandTextBuilder CreateDdlTextBuilder(string ddlStatement)
+        public static SpannerCommandTextBuilder CreateDdlTextBuilder(string ddlStatement, string[] extraDdlStatements)
```
Comment (jskeet) [inline apis/Google.Cloud.Spanner.Data/Google.Cloud.Spanner.Data/SpannerCommandTextBuilder.cs]: Would this make sense to be a parameter array? (As it is for CreateDdlCommand)

File: apis/Google.Cloud.Spanner.Data/Google.Cloud.Spanner.Data.IntegrationTests/Admin.cs (hunk 1 above)
Comment (benwulfe) [reply apis/Google.Cloud.Spanner.Data/Google.Cloud.Spanner.Data.IntegrationTests/Admin.cs]: yep

File: apis/Google.Cloud.Spanner.Data/Google.Cloud.Spanner.Data.IntegrationTests/Admin.cs (hunk 2 above)
Comment (benwulfe) [reply apis/Google.Cloud.Spanner.Data/Google.Cloud.Spanner.Data.IntegrationTests/Admin.cs]: yes.  this feature is supported natively by the createdb admin endpoint.

File: apis/Google.Cloud.Spanner.Data/Google.Cloud.Spanner.Data/SpannerCommand.cs (hunk 3 above)
Comment (benwulfe) [reply apis/Google.Cloud.Spanner.Data/Google.Cloud.Spanner.Data/SpannerCommand.cs]: hmm create database cannot be one of the extra ddl statements (it must be the first one if specified).  I will make that clear in the docs.

File: apis/Google.Cloud.Spanner.Data/Google.Cloud.Spanner.Data/SpannerCommandTextBuilder.cs (hunk 5 above)
Comment (benwulfe) [reply apis/Google.Cloud.Spanner.Data/Google.Cloud.Spanner.Data/SpannerCommandTextBuilder.cs]: makes sense

**Comment by jskeet** [comment]
(LGTM other than the nits btw, in case that wasn't clear :)

**Comment by benwulfe** [comment]
thanks Jon.  note that coverage is done via integration tests, so wont show up in codecov.
on the bright side, codacy finally likes me.

**Pull request merged**